controller/src/main/java/org/dswarm/controller/resources/resource/DataModelsResource.java

```diff
@@ -997,4 +911,58 @@ private WebTarget target(final String... path) {
 		return optionalValue;
 	}
 
+	private static final class StreamingDataObserver implements Observer<Tuple<String, JsonNode>> {
+		private final ObjectMapper objectMapper;
+		private final String pojoClassName;
+		private final AsyncResponse asyncResponse;
+		private final String uuid;
+		private final ObjectNode json;
+
+		private boolean hasData;
+
+		public StreamingDataObserver(final String uuid, final ObjectMapper objectMapper, final String pojoClassName, final AsyncResponse asyncResponse) {
+			this.uuid = uuid;
+			this.objectMapper = objectMapper;
+			this.pojoClassName = pojoClassName;
+			this.asyncResponse = asyncResponse;
+			json = objectMapper.createObjectNode();
+		}
+
+		@Override
+		public void onCompleted() {
+
+			if (!hasData) {
+
+				DataModelsResource.LOG.debug("couldn't find data for data model with uuid '{}'", uuid);
+				asyncResponse.resume(Response.status(Status.NOT_FOUND).build());
+			} else {
+
+				try {
+					final String jsonString = serializesObject(json, objectMapper, pojoClassName);
```

Comment:
so, this is the point, where we could continue streaming, or? - i.e. write the serialized objects to an output stream (instead of a string). anyway, I think this is some thing for later ;)